ts/core/MmlTree/MmlNodes/mtext.ts

```diff
@@ -51,11 +51,18 @@ export class MmlMtext extends AbstractMmlTokenNode {
   }
 
   /**
-   * <mtext> is always space-like according to the spec
+   * <mtext> is always space-like according to the spec,
+   * but we make it so only if it contains only spaces and
+   * doesn't have certain attributes
+   *
    * @override
    */
   public get isSpacelike() {
-    return true;
+    const attributes = this.attributes;
+    const spaces = !!this.getText().match(/^\s*$/);
+    return spaces && (attributes.getExplicit('mathbackground') === undefined &&
+                      attributes.getExplicit('background') === undefined &&
+                      attributes.getExplicit('style') === undefined);
```

Comment:
Should those attributes go into a dedicated constant, in case we need more in the future?

Reply:
There are a couple of places where this type of check is being made, and I was thinking about making a service function to check them.  I'll look into that.

Reply:
I have added `hasOneOf()` to the `Attributes` object to handle this, and made lists for the styles to use for `space` and `mtext` elements.  I also added a `hasExplicit()` method to the `Attributes` object to make checking for explicit attributes more natural, and `unset()` to make removing an attribute easier.  I edited a number of other files to use these new functions.  So there are a bunch of new edits here.  Do you want to review again?